From 4613274f20076d715fd6489e54a8bf266baa367f Mon Sep 17 00:00:00 2001
From: Nikhil Badola <nikhil.badola@freescale.com>
Date: Tue, 9 Dec 2014 19:00:32 +0530
Subject: [PATCH 272/767] usb: gadget: Turn on cache snooping hardware in usb
 gadget

Turn on cache snooping hardware in usb gadget for PPC 64-bit Kernel

Signed-off-by: Nikhil Badola <nikhil.badola@freescale.com>
Signed-off-by: yinbo.zhu <yinbo.zhu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/usb/gadget/udc/fsl_udc_core.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/gadget/udc/fsl_udc_core.c b/drivers/usb/gadget/udc/fsl_udc_core.c
index 4d365257aa9f..d24a63c92ccd 100644
--- a/drivers/usb/gadget/udc/fsl_udc_core.c
+++ b/drivers/usb/gadget/udc/fsl_udc_core.c
@@ -332,6 +332,12 @@ static int dr_controller_setup(struct fsl_udc *udc)
 	}
 #endif
 
+#if (defined(CONFIG_PPC32) || defined(CONFIG_PPC64)) && \
+!defined(CONFIG_NOT_COHERENT_CACHE)
+/* Turn on cache snooping hardware, since some PowerPC platforms
+ * wholly rely on hardware to deal with cache coherent.
+ */
+
 #if !defined(CONFIG_NOT_COHERENT_CACHE)
 	/* Turn on cache snooping hardware, since some PowerPC platforms
 	 * wholly rely on hardware to deal with cache coherent. */
-- 
2.17.0

